usr/src/test/zfs-tests/tests/functional/cli_root/zpool/zpool_003_pos.ksh

```diff
@@ -57,14 +58,20 @@ fi
 
 log_mustnot zpool freeze fakepool
 
-# Remove corefile possibly left by previous failing run of this test.
-[[ -f core ]] && log_must rm -f core
+#preparation work for testing
+corepath=$TESTDIR/core
+if [[ -d $corepath ]]; then
+	rm -rf $corepath
+fi
+mkdir $corepath
```

Comment:
The previous codepath would remove `corepath` if it existed as a regular file.

The new codepath will remove `corepath` if it exists as a directory.

The `mkdir $corepath` will fail if a file exists with this name.

Thanks

Reply:
Thanks for the feedback.  I added an 's' to the end of the path which should address that issue.